packages/viewer/src/modules/menu/components/advancedTools/ImportCatalogue/useProviders.ts

```diff
@@ -1,32 +1,43 @@
+import type { ComputedRef, Ref } from 'vue'
+
 import log from '@swissgeo/log'
 import { computed, ref } from 'vue'
 
 import providersJson from '@/modules/menu/components/advancedTools/ImportCatalogue/external-providers.json'
 
 class Provider {
-    constructor(url, emphasize = '') {
+    url: string
+    htmlDisplay: string
+    emphasize: string
+
+    constructor(url: string, emphasize: string = '') {
         this.url = url
         this.htmlDisplay = url.replace(/access_key=.*/, '')
         this.emphasize = emphasize
     }
 
-    getBaseUrl() {
+    getBaseUrl(): string | undefined {
         try {
             const urlObj = new URL(this.url)
             return urlObj.origin
         } catch (e) {
-            log.error('Invalid URL:', this.url, e)
-            return null
+            log.error({
+                title: 'Invalid URL',
+                message: [this.url, e instanceof Error ? e.message : String(e)],
```

Comment:
```suggestion
                message: [this.url, e]
```
should be fine, type wise